### PR TITLE
[Woo POS] Add collapsed cart view. Update item grid to list

### LINKED ISSUE #12980: [Woo POS] Collapsed cart view when empty
From i3, the cart starts in a greyed-out state which is not completely visible, just becoming visible when an operator adds products to the cart:
Screenshot 2024-06-06 at 15 19 02

Logging this issue so we can explore this behaviour. Something to take into account is localization, since may cut-off or expand the ideal cart width.

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #12980

Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-06-17 at 10 43 13

## Description
- Adds a `CollapsedCartView`,  which will render when the cart is empty. Switching for the existing `CartView` when items are added to the cart.
- Removes the placeholders for `SearchView` and `FilterView`, since these fell out of scope for the MVP.
- Adds a `ItemListView` to display all products on a list manner, rather than a grid. This is mostly just a copy of the pre-existing grid.

## Testing:
- In POS mode, observe that the items are now rendered as a list, and the cart is initially hidden.
- Upon adding products to the cart, the cart will show up. 

There are some more improvements to be made, like the item cards do not longer have a `+` button but the whole thing seems to be tappable, we've left this for the next PR.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift [hunk 1]
```diff
@@ -11,4 +12,26 @@ final class PointOfSaleDashboardViewModelTests: XCTestCase {
         // Then
         XCTAssertTrue(viewModel.itemsInCart.isEmpty)
     }
+
+    func test_cart_is_collapsed_when_empty_then_not_collapsed_when_has_items() {
+        // Given
+        let viewModel = PointOfSaleDashboardViewModel(items: POSItemProviderPreview().providePointOfSaleItems(),
```
Comment (iamgabrielma) [inline WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift:18]: I've used the `POSItemProviderPreview` class here following the same pattern as existing tests, but should we either should use something else, or rename this to clarify it's used as a mock?

File: WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift [hunk 2]
```diff
@@ -1,14 +1,46 @@
 import XCTest
+@testable import struct Yosemite.POSProduct
 @testable import WooCommerce
 
 final class PointOfSaleDashboardViewModelTests: XCTestCase {
 
+    private var sut: PointOfSaleDashboardViewModel!
+    private var cardPresentPaymentService: CardPresentPaymentPreviewService!
```
Comment (iamgabrielma) [inline WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift:8]: Is `CardPresentPaymentPreviewService` something we'll use for both previews and as a mock? Or we plan to separate these? If we don't maybe we should rename this to `CardPresentPaymentMockService` or similar 🤔

**Comment by bozidarsevo** [comment]
> * which will render when the cart is empty

did you mean "is not empty"?

File: WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift (hunk 2 above)
Comment (bozidarsevo) [reply WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift:8]: maybe @joshheald knows a bit more?

**Review by bozidarsevo — APPROVED** [review]
Looking forward to all the next tweaks and updates that makes us get closer to final design 😎

**Comment by iamgabrielma** [comment]
Thanks for the review!

> Looking forward to all the next tweaks and updates that makes us get closer to final design 😎

Yeah, agree! With the latest high fidelity designs up it feels we can start to work on these without reverting/re-working too much 👯

**Comment by bozidarsevo** [comment]
> > * which will render when the cart is empty
> 
> did you mean "is not empty"?

Now I see I missread what view is shown 😅👍

File: WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift (hunk 2 above)
Comment (joshheald) [reply WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift:8]: I only added it to make it easier to do previews. It's the simplest possible implementation of `CardPresentPaymentFacade`, and importantly has no dependencies.

It kinda works as a test double as well, but it's not really got some of the stuff we're likely to want when testing, e.g. spy variables, the opportunity to set test state/mocked responses.

Adding those would make it worse at its primary job (being a simple structure to make preview creation easy) so we wouldn't want to add that, or rename it to make the role of the class less clear.

Adding a new mock is the way to go here, IMO.

Comment (iamgabrielma) [reply WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift:8]: Sounds good, thanks for the clarification. I've logged it here for now: https://github.com/woocommerce/woocommerce-ios/issues/13077

**Pull request merged**